src/app/services/technical-record/technical-record.service.ts

```diff
@@ -173,13 +173,18 @@ export class TechnicalRecordService {
           return true;
         }
 
+        const allTechRecords = vehicleTechRecord.flatMap(record => record.techRecord);
+        const allTechRecordsArchived = allTechRecords.every(record => record.statusCode === StatusCodes.ARCHIVED);
+        if (allTechRecordsArchived) {
+          return true;
+        }
+
         if (searchType === SEARCH_TYPES.VRM) {
           const allVrms = vehicleTechRecord.flatMap(record => record.vrms);
           const primaryVRMFound = allVrms.some(vrm => vrm.isPrimary && vrm.vrm == valueToCheck);
           return !primaryVRMFound;
         }
-        const allTechRecords = vehicleTechRecord.flatMap(record => record.techRecord);
-        return allTechRecords.every(record => record.statusCode === StatusCodes.ARCHIVED);
+        return allTechRecordsArchived;
```

Comment:
```typescript
        if (
          vehicleTechRecord
            .flatMap(vehicle => vehicle.techRecord)
            .every(record => record.statusCode === StatusCodes.ARCHIVED)
        ) {
          return true;
        }

        if (searchType === SEARCH_TYPES.VRM) {
          return !vehicleTechRecord
            .flatMap(record => record.vrms)
            .some(vrm => vrm.isPrimary && vrm.vrm === valueToCheck);
        }
        
        return false;
```

Why am I suggesting this change?
1. No need to store things in variables because they are not used anywhere else and the conditional logic is not difficult to understand. All you are doing is flattening the records and either verifying the `statusCode` or the `vrm`.
2. Returning `allTechRecordsArchived` is confusing and incorrect at the end, because you will always return `true` on line 179 if `allTechRecordsArchived === true`. Thus, the end return will always be `false`.
3. No need to early exit by checking `vehicleTechRecord.length === 0` as `.flatMap().every()` will return `true` on empty arrays.

You can verify this by running the following on MDN:
```typescript
const array = [];
console.log(array.flatMap(num => num.yolo).every(num => num.jinx === 'jinx'));
```

Reply:
Don't think so, using every would mean it'd only be counted as not unique if all the VRMs it finds are primary and equal the value, as opposed to at least one matching that condition with the some.

Reply:
I think @mcadenas-bjss wins here, I admit defeat.

Reply:
Yeah, but using every suggests that for a VRM to not be unique, **all** of the existing VRMs it finds have to be primary and equal to the VRM we're amending to, as opposed to checking for at least one with some.